client-app/src/pages/account/views/account-invoices/account-invoices.ts

```diff
@@ -1,66 +1,89 @@
 import { pageSizes, invoicesStatuses, baseUrl, storeName, locale } from "@common/constants";
-
+import { PaymentSearchQuery } from '@common/models/search/extensions/payment-search-query';
+import { QueryBuilder } from '@common/services/query-builder.service';
+import "@common/models/search/extensions/payment-search-criteria";
 
 const invoicesListModule = namespace('invoicesListModule');
 
 @Component({
   components: {
     InvoicesFilter
+  },
+  beforeRouteUpdate: function (to: Route, from: Route, next: (to?: RawLocation | false | ((vm: AccountInvoices) => any) | void) => void) {
+    (this as AccountInvoices).routeChanged(to);
+    next();
   }
 })
 export default class AccountInvoices extends Vue {
-  @invoicesListModule.Getter("InvoicesList")
-  private invoicesList!: InvoicesList;
-
   @invoicesListModule.Getter("isLoading")
   private isLoading!: boolean;
 
-  @invoicesListModule.Action(FETCH_INVOICES)
-  private fetchInvoices!: () => InvoicesList;
+  @invoicesListModule.Getter("columns")
+  private columns!: boolean;
+
+  @invoicesListModule.Getter("searchCriteria")
+  private searchCriteria!: IPaymentSearchCriteria;
+
+  @invoicesListModule.Action(SET_INVOICES_SEARCH_CRITERIA)
+  private setSearchCriteria!: (searchCriteria: IPaymentSearchCriteria) => void;
 
-  @invoicesListModule.Action(SET_INVOICES_LIST_CONFIG)
-  private setListConfig!: (listConfig: InvoicesListConfig) => void;
+  @invoicesListModule.Getter("invoices")
+  private invoices!: IPaymentSearchResult;
 
   downloadIcon = faSave;
 
   pageSizes = pageSizes;
 
   availableInvoicesStatuses = invoicesStatuses;
 
+  queryBuilder = new QueryBuilder(PaymentSearchCriteria, PaymentSearchQuery);
+
   mounted() {
-    this.fetchInvoices();
+    this.routeChanged(this.$route);
+  }
+
+  routeChanged(route: Route) {
+    const searchCriteria = this.queryBuilder.parseQuery(route.query);
+    this.setSearchCriteria({
+      ...this.searchCriteria,
+      ...searchCriteria
+    });
   }
 
-  pageChanged(page: number) {
-    this.setListConfig({ ...this.invoicesList.listConfig, pageNumber: page });
+  pageChanged(pageNumber: number) {
+    this.searchCriteriaChanged({ ...this.searchCriteria, pageNumber });
   }
 
   pageSizeChanged(pageSize: number) {
-    this.setListConfig({ ...this.invoicesList.listConfig, pageNumber: 1, pageSize: pageSize });
+    this.searchCriteriaChanged({ ...this.searchCriteria, pageNumber: 1, pageSize });
   }
 
   sortChanged(ctx: BvTableCtxObject) {
     const sortDirection = ctx.sortDesc ? "desc" : "asc";
     const sortExpression = `${ctx.sortBy}:${sortDirection}`;
-    const listConfig = { ...this.invoicesList.listConfig, pageNumber: 1 };
-    listConfig.filters = { ...this.invoicesList.listConfig.filters, sort: sortExpression };
-    this.setListConfig(listConfig);
+    const searchCriteria = { ...this.searchCriteria, pageNumber: 1, sort: sortExpression };
+    this.searchCriteriaChanged(searchCriteria);
   }
 
   checkActivePageSize(pageSize: number) {
```

Comment:
1. check - it can be something large when you using 'is' it is 1. shorter, 2. means short checking
2. I presume what code "active="isActivePageSize(pageSize)"" more pure than "active="checkActivePageSize(pageSize)""

Reply:
Same as above